atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -683,6 +695,34 @@ private void extractLocalWritesForRow(@Output Map<Cell, byte[]> result,
                 });
     }
 
+    @Override
+    public <T> Stream<T> getRanges(
+            final TableReference tableRef,
+            Iterable<RangeRequest> rangeRequests,
+            int concurrencyLevel,
+            Function<BatchingVisitable<RowResult<byte[]>>, T> visitableProcessor) {
```

Comment:
i feel like we should also pass the range request that resulted in the visitable into the processor. currently you can associate them based on the ordering (i.e., the ith batch corresponds to the ith range request), but it'd be nicer to just get it directly

Reply:
Fair point. Will do that.